File: from-0001-to-0100/0080-remove-duplicates-from-sorted-array-ii/solution.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        fast = 1
        slow = 1
        prev = nums[0]
        count = 1
        
        while fast < len(nums):
            if nums[fast] != prev:
                prev = nums[fast]
                nums[slow] = prev
                slow += 1
                count = 1
            else:
                count += 1
                if count <= 2:
                    nums[slow] = prev
                    slow += 1
                    count += 1                    
                
            fast += 1
            
        return slow

File: from-0001-to-0100/0080-remove-duplicates-from-sorted-array-ii/solution.go
func removeDuplicates(nums []int) int {
    nDupl := 1
    prev := nums[0]
    tail := 1
    for i := 1; i < len(nums); i ++ {
        if nums[i] == prev {
            nDupl += 1
            if nDupl <= 2 {
                nums[tail] = nums[i]
                tail += 1
            }
        } else {
            nDupl = 1
            nums[tail] = nums[i]
            tail += 1
        }
        prev = nums[i]
    }
    return tail
}